chore: setup rpc node docs (#235)
chainflip-io/chainflip-docs: pages/mainnet/running-an-rpc-node.mdx

---

title: Running an RPC node

> description: This guide describes how to download and run Chainflip's validator software and run an RPC node.

Reviewer: ```suggestion
description: This guide describes how to download and run a Chainflip RPC node.
```
best not to potentially confuse the issue by mentioning "validator"

---

import {Callout} from "@/components";

# Running an RPC node

> Running an RPC node for Chainflip is a relatively simple process. You can decide to run an archive node or just a simple RPC node.

Reviewer: A small description about what the difference is would be useful. Like 
"A simple RPC node will only store recent state, how much state you wish to store can be configured. This utilises less disk space than an archive node. An archive node, however, stores the state from the beginning of the chain's lifetime and so is necessary for historical state queries.

Reviewer 2: further nitpick: The current framing seems to suggest a binary choice between archive/non-archive - IMO this puts too much emphasis on archive mode. We should assume here that 90% of users just want to spin up a vanilla rpc node. We can mention archive mode in its own section, as a special case that might be useful for eg. block explorers. I don't think it needs a special mention in the intro.

</Callout>

```bash copy
sudo mkdir -p /etc/apt/trusted.gpg.d
gpg --export 4E506212E4EF4E0D3E37E568596FBDCACBBCDD37 | sudo tee /etc/apt/trusted.gpg.d/chainflip-mainnet.gpg
echo "deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/chainflip-mainnet.gpg] https://pkgs.chainflip.io/ubuntu/ jammy berghain" | sudo tee /etc/apt/sources.list.d/chainflip-berghain-mainnet.list
```

## Installing The Packages

To install, run the following commands:

### Archive Node

```bash copy
sudo apt update
sudo apt install -y chainflip-archive-node
```

On successful installation, you can start the archive node by running:

```bash copy
sudo systemctl start chainflip-archive-node
```

Check that the node is running by running:

```bash copy
sudo systemctl status chainflip-archive-node
```

You can check that the node is syncing by running:

```bash copy
curl -H "Content-Type: application/json" -s -d '{"id":1, "jsonrpc":"2.0", "method": "system_health","id":1}' http://localhost:9944 -v
```

### RPC Node

```bash copy
sudo apt update
sudo apt install -y chainflip-rpc-node
```

On successful installation, you can start the RPC node by running:

```bash copy

> sudo systemctl start chainflip-rpc-node

Reviewer: How far back is the pruning on an RPC node? Would be good to provide this, and also provide how the user can configure it

```

Check that the node's status by running:

```bash copy
sudo systemctl status chainflip-rpc-node
```

You can check that the node is syncing by running:

```bash copy
sudo journalctl -u chainflip-rpc-node -f
```

and check the logs for the following:

```text
Nov 14 13:38:58 berghain-validator chainflip-node[1908]: 2023-11-14 13:38:58 ⏩ Warping, Downloading state, 7.02 Mib (4 peers), best: #0 (0x8b8c…6eb9), finalized #0 (0x8b8c…6eb9), ⬇ 372.3kiB/s ⬆ 0.2kiB/s
```

# Docker Installation

<Callout type="warning">
If you have done the **APT Installation** above, you don't need to follow the **Docker Installation** below.
</Callout>

If you are planning to run a validator using a container orchestration system like Kubernetes you can use our pre-built images on [Docker Hub](https://hub.docker.com/u/chainfliplabs).

```bash copy
docker pull chainfliplabs/chainflip-node:berghain-1.3.2
```

## Starting up the node in Docker

> To start the node, you would need the following options passed to the `chainflip-node` binary:

Reviewer: ```suggestion
To start the node, you would need the following options passed to the `chainflip-node` binary:
```

To a user unfamiliar with the binaries, this could seem like a mistake, this is the first time `chainflip-node` has been mentioned, the rest referred to RPC so this could be confusing. A small note about this would help.
